File: src/main/java/de/danielmaile/aether/worldgen/dungeon/OuterPartType.java
package de.danielmaile.aether.worldgen.dungeon;

import de.danielmaile.aether.worldgen.PrefabType;

public enum OuterPartType
{
    S(PartType.S),
    N(PartType.N),
    W(PartType.W),
    E(PartType.E),

    SN(PartType.SN),
    WS(PartType.WS),
    ES(PartType.ES),
    WN(PartType.WN),
    EN(PartType.EN),
    EW(PartType.EW),

    EWS(PartType.EWS),
    WSN(PartType.WSN),
    ESN(PartType.ESN),
    EWN(PartType.EWN),

    EWSN(PartType.EWSN);

    public PartType getType()
    {
        return type;
    }

    private final PartType type;

    public PrefabType getPrefabType()
    {
        return prefabType;
    }

    private final PrefabType prefabType;

    OuterPartType(PartType type)
    {
        this(type, null);
    }

    OuterPartType(PartType type, PrefabType prefabType)
    {
        this.type = type;
        this.prefabType = prefabType;
    }

    public Connection getConnection()
    {
        return this.type.getConnection();
    }
}

File: src/main/java/de/danielmaile/mpp/aether/world/dungeon/OuterPartType.kt
package de.danielmaile.mpp.aether.world.dungeon

import de.danielmaile.mpp.aether.world.PrefabType

enum class OuterPartType(val type: PartType, val prefabType: PrefabType?) {
    S(PartType.S, null),
    N(PartType.N,null),
    W(PartType.W, null),
    E(PartType.E, null),

    SN(PartType.SN, null),
    WS(PartType.WS, null),
    ES(PartType.ES, null),
    WN(PartType.WN, null),
    EN(PartType.EN, null),
    EW(PartType.EW, null),

    EWS(PartType.EWS, null),
    WSN(PartType.WSN, null),
    ESN(PartType.ESN, null),
    EWN(PartType.EWN, null),

    EWSN(PartType.EWSN, null);

    fun getConnection(): Connection {
        return type.connection
    }
}